addons/website/models/website.py

```diff
@@ -1751,7 +1750,13 @@ def _trigram_enumerate_words(self, search_details, search, limit):
             best_similarity = sql.SQL('GREATEST({similarities})').format(
                 similarities=sql.SQL(', ').join(similarities)
             )
-
+            # This method uses hardcoded sql query, it needs filter regarding performance.
+            # In this case for the is_published field as customer don't have access to
+            # unpublished objects.
+            if 'is_published' in model._fields and not self.env.user.has_group('base.group_user'):
+                where_clause = sql.SQL("WHERE is_published")
+            else:
+                where_clause = sql.SQL("WHERE True")
```

Comment:
This can just be an empty string before the `if`, no need to to add `WHERE True`, isn't it? It doesn't bring anything to the query (it's probably a leftover of the previous implementation which had overridable hooks)